Source/Common/pch_common.h

```diff
@@ -128,6 +128,7 @@ HC_DECLARE_TRACE_AREA(WEBSOCKET);
     catch (...) { ::xbox::httpclient::detail::UnknownExceptionToResult(file, line); return errCode; }
 
 #define RETURN_IF_PERFORM_CALLED(call) if (call->performCalled) return E_HC_PERFORM_ALREADY_CALLED;
+#define TO_ULL(x) static_cast<unsigned long long>(x)
```

Comment:
Could use a templated function instead of a macro if you want. Adjust the `enable_if` as needed. (There's probably an even better way to write the condition. I was going for "no truncation".)

```suggestion
template <typename T>
typename std::enable_if<
    std::is_unsigned<T>::value && sizeof(T) <= sizeof(unsigned long long),
    unsigned long long>::type
ToUll(const T val)
{
    return val;
}
```

Reply:
There shouldn't be truncation as discussed here: https://stackoverflow.com/questions/14073877/printf-and-llu-vs-lu-on-os-x cause on some platforms its unsigned long long which is fine, and some platforms its unsigned long, so casting to unsigned long long is fine. 

I'd prefer not to add complex templates for a small issue as this.